### ISSUE TITLE
stardist: config file doesn't exist

### ISSUE THREAD

**Issue by GBeattie** [stardist/stardist#285]
Hi, great looking package, just hitting a snag with the stardist step, I'm following the notebook exactly with no issues till now, but stardist runs a step where it downloads from it's repository (not present in the example notebook) then pulls an error about a missing config.json. I'm running Python 3.11.8, with tensorflow installed.

```
b2c.stardist(image_path="stardist/he.tiff", 
             labels_npz_path="stardist/he.npz", 
             stardist_model="2D_versatile_he", 
             prob_thresh=0.01
            )

Found model '2D_versatile_he' for 'StarDist2D'.
Downloading data from https://github.com/stardist/stardist-models/releases/download/v0.1/python_2D_versatile_he.zip
5294730/5294730 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[9], line 1
----> 1 b2c.stardist(image_path="stardist/he.tiff", 
      2              labels_npz_path="stardist/he.npz", 
      3              stardist_model="2D_versatile_he", 
      4              prob_thresh=0.01
      5             )

File ~/anaconda3/lib/python3.11/site-packages/bin2cell/bin2cell.py:120, in stardist(image_path, labels_npz_path, stardist_model, block_size, min_overlap, context, **kwargs)
    118 img = normalize(img)
    119 #use pretrained stardist model
--> 120 model = StarDist2D.from_pretrained(stardist_model)
    121 #will need to specify axes shortly, which are model dependent
    122 if stardist_model == "2D_versatile_he":
    123     #3D image, got the axes YXC from the H&E model config.json

File ~/anaconda3/lib/python3.11/site-packages/csbdeep/models/base_model.py:79, in BaseModel.from_pretrained(cls, name_or_alias)
     77 try:
     78     get_model_details(cls, name_or_alias, verbose=True)
---> 79     return get_model_instance(cls, name_or_alias)
     80 except ValueError as e:
     81     if name_or_alias is not None:

File ~/anaconda3/lib/python3.11/site-packages/csbdeep/models/pretrained.py:107, in get_model_instance(cls, key_or_alias)
    105 def get_model_instance(cls, key_or_alias):
    106     path = get_model_folder(cls, key_or_alias)
--> 107     model = cls(config=None, name=path.stem, basedir=path.parent)
    108     model.basedir = None # make read-only
    109     return model

File ~/anaconda3/lib/python3.11/site-packages/stardist/models/model2d.py:293, in StarDist2D.__init__(self, config, name, basedir)
    291 def __init__(self, config=Config2D(), name=None, basedir='.'):
    292     """See class docstring."""
--> 293     super().__init__(config, name=name, basedir=basedir)

File ~/anaconda3/lib/python3.11/site-packages/stardist/models/base.py:223, in StarDistBase.__init__(self, config, name, basedir)
    222 def __init__(self, config, name=None, basedir='.'):
--> 223     super().__init__(config=config, name=name, basedir=basedir)
    224     threshs = dict(prob=None, nms=None)
    225     if basedir is not None:

File ~/anaconda3/lib/python3.11/site-packages/csbdeep/models/base_model.py:106, in BaseModel.__init__(self, config, name, basedir)
    103 if config is not None:
    104     # config was provided -> update before it is saved to disk
    105     self._update_and_check_config()
--> 106 self._set_logdir()
    107 if config is None:
    108     # config was loaded from disk -> update it after loading
    109     self._update_and_check_config()

File ~/anaconda3/lib/python3.11/site-packages/csbdeep/models/base_model.py:32, in suppress_without_basedir.<locals>._suppress_without_basedir.<locals>.wrapper(*args, **kwargs)
     30     warn is False or warnings.warn("Suppressing call of '%s' (due to basedir=None)." % f.__name__)
     31 else:
---> 32     return f(*args, **kwargs)

File ~/anaconda3/lib/python3.11/site-packages/csbdeep/models/base_model.py:146, in BaseModel._set_logdir(self)
    144             raise ValueError('Invalid attributes in loaded config: ' + ', '.join(invalid_attr))
    145     else:
--> 146         raise FileNotFoundError("config file doesn't exist: %s" % str(config_file.resolve()))
    147 else:
    148     if self.logdir.exists():

FileNotFoundError: config file doesn't exist: /Users/gordonbeattie/.keras/models/StarDist2D/2D_versatile_he/config.json
```

Thanks in advance for any assistance!

**Comment by maweigert** [comment]
Hi, 

you could check whether the keras model cache directory contains a `config.json`:

`ls /Users/gordonbeattie/.keras/models/StarDist2D/2D_versatile_he/`


otherwise I would open an issue at [bin2cell](https://github.com/Teichlab/bin2cell) as this is where the error seems to come from

**Comment by GBeattie** [comment]
Thanks for the quick response, managed to fix it, the model was extracted to a separate directory, "2D_versatile_he_extracted", which contained the .jsons and the .h5, I just moved these to the parent and now it's running, thanks!

```
cd /Users/gordonbeattie/.keras/models/StarDist2D/2D_versatile_he/
ls
2D_versatile_he.zip		2D_versatile_he_extracted
cd 2D_versatile_he_extracted
ls
config.json	thresholds.json	weights_best.h5
mv * ..
```

**Comment by arthursw** [comment]
This is due to `csbdeep` 0.8.0 which were out-of-date with the keras v3.8.0 ([this commit](https://github.com/keras-team/keras/commit/dcefb139863505d166dd1325066f329b3033d45a) made the get_file function return the extracted path, so csbdeep.get_model_folder must return this path, not its parent). 

It was fixed in `csbdeep` 0.8.1. 

Thus, stardist should require `csbdeep >= 0.8.1` instead of `csbdeep >= 0.8.0`.
